blog/2024-11-11-package.md

```diff
@@ -0,0 +1,136 @@
+
+The `lib` directory should be placed alongside the `src` directory, organized as follows:
+
+```
+├── .
+│   ├── src/
+│   │   ├── lib/
+│   │   │   ├── private/
+│   │   │   │   └── AbstractLibraryFile.lf
+│   │   │   ├── LibraryFile_1.lf
+│   │   │   └── LibraryFile_2.lf
+│   │   └── Foo.lf
+...
+```
+
+In LF, there is no visibility control, meaning you cannot mark a reactor or file as `private` as in other programming languages. The `private` directory serves this purpose by containing files meant solely for internal package use. Files in this directory are hidden in the Package Explorer within the VS Code extension, making them inaccessible to other packages. The `private` directory is useful for implementing design patterns like the Abstract Factory Pattern, which provides an interface for creating families of related objects without specifying concrete classes. For example, a developer might define an `AbstractLibraryFile.lf` containing abstract reactors and implement them in various ways within different files under the `lib` directory.
+
+### Must-Follow Guidelines
+
+When creating a new package, ensure that you follow these guidelines:
+
+- **README.md**: Include a `README.md` file in the root directory. This should provide an overview of the package, its purpose, a description of the LF files in the `lib/` folder, and relevant user information. The `README.md` is essential for publishing your package to the community repository and must be kept up to date.
+
+- **File Naming**: Use descriptive names for files in the `lib/` directory that reflect their purpose. For example, name a file containing a reactor for different sensors as `Sensor.lf`.
+
+- **Example LF Files**: Include example LF files in the `src/` directory to demonstrate how to use the reactors in the `lib/` directory. These examples should be well-documented with comments explaining the purpose and usage of each reactor.
+
+## Configuring the `Lingo.toml` File
+
+The `Lingo.toml` file is a configuration file that provides essential information about your package, such as its name, version, and dependencies. It ensures that your package is recognized and installed correctly by the Lingo Package Manager. The `Lingo.toml` file should include the following sections:
+
+- **`[package]`**: Specify the package name, description, and version.
+
+- **`[lib]`**: Define the package name, the target language used to implement the code, the main reactor (which could be one of the examples from the `src/` directory), and the platform used to build the package.
+
+Here's an example of a `Lingo.toml` file:
+
+```toml
+[package]
+name = "PackageName"
+version = "0.3.1"
+
+[properties]
+
+[lib]
+name = "PackageName"
+main = "./src/Main.lf"
```

Comment:
It is not clear to me what `main` should be for a library. Typically, a library will have multiple example programs illustrating the use of the library.  What file should this point to?